src/modules/fancyzones/lib/trace.cpp

```diff
@@ -64,6 +65,99 @@ void Trace::FancyZones::OnKeyDown(DWORD vkCode, bool win, bool control, bool inM
         TraceLoggingBoolean(inMoveSize, "InMoveSize"));
 }
 
+void Trace::FancyZones::DataChanged() noexcept
+{
+    const JSONHelpers::FancyZonesData& data = JSONHelpers::FancyZonesDataInstance();
+    int appsHistorySize = data.GetAppZoneHistoryMap().size();
+    const auto& customZones = data.GetCustomZoneSetsMap();
+    const auto& devices = data.GetDeviceInfoMap();
+
+    INT32* customZonesArray = new (std::nothrow) INT32[customZones.size()];
+    INT32* templateZonesArray = new (std::nothrow) INT32[devices.size()];
```

Comment:
Use smart-pointers here or `delete[]` these in the end

Reply:
Thank you! Fixed